### PR TITLE
call UpdateNode during Refresh

### LINKED ISSUE #1663: Tree failing to call UpdateNode on manual refresh
Hi,
is it possible to change a node label text in a tree widget?

thank you)

### PR DESCRIPTION
**Pull request by andydotxyz**
Resolve issue where tree did not refresh from data using UpdateNode

Fixes #1663

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

**Review by Jacalz — APPROVED** [review]
Looks good to me, but it is probably best if @stuartmscott have a look as well :)

File: widget/tree.go [hunk 1]
```diff
@@ -694,7 +695,9 @@ func (r *treeNodeRenderer) Objects() (objects []fyne.CanvasObject) {
 
 func (r *treeNodeRenderer) Refresh() {
 	if c := r.treeNode.content; c != nil {
-		c.Refresh()
+		if f := r.treeNode.tree.UpdateNode; f != nil {
+			f(r.treeNode.uid, r.treeNode.isLeaf, c)
```
Comment (stuartmscott) [inline widget/tree.go:699]: The second parameter of `UpdateNode` is `isBranch`

File: widget/tree.go [hunk 2]
```diff
@@ -782,6 +786,7 @@ func newLeaf(tree *Tree, content fyne.CanvasObject) (l *leaf) {
 		&treeNode{
 			tree:    tree,
 			content: content,
+			isLeaf:  false,
```
Comment (stuartmscott) [inline widget/tree.go:789]: Copy/Pasta error - should be true

File: widget/tree.go [hunk 3]
```diff
@@ -586,6 +586,7 @@ type treeNode struct {
 	depth   int
 	hovered bool
 	icon    fyne.CanvasObject
+	isLeaf  bool
```
Comment (stuartmscott) [inline widget/tree.go:589]: Let's keep consistent with the rest of Tree and use `isBranch`

**Review by stuartmscott — CHANGES_REQUESTED** [review]

File: widget/tree.go (hunk 1 above)
Comment (andydotxyz) [reply widget/tree.go:699]: good catch thanks, will fix the logic

File: widget/tree.go (hunk 3 above)
Comment (andydotxyz) [reply widget/tree.go]: yeah, as above, good catch

File: widget/button_test.go [hunk 4]
```diff
@@ -65,6 +65,10 @@ func TestButton_Tapped(t *testing.T) {
 }
 
 func TestButton_Disable(t *testing.T) {
+	app := test.NewApp()
+	defer test.NewApp()
+	app.Settings().SetTheme(theme.LightTheme())
```
Comment (stuartmscott) [inline widget/button_test.go:70]: This is not related to fixing #1663, please remove

Comment (andydotxyz) [reply widget/button_test.go:70]: Are you sure? It seemed to start failing when I added the tree test, hence the commit message "Resolve flakey button test impacted by this tree change".

Comment (andydotxyz) [reply widget/button_test.go]: Seems like this was broken on release branch and we didn't notice.
Rebase on top of the fix, so it's no longer part of this PR diff.

**Review by stuartmscott — APPROVED** [review]

**Pull request merged**